crates/bevy_graph/src/algos/bfs.rs

```diff
@@ -0,0 +1,135 @@
+use std::collections::VecDeque;
+
+use hashbrown::HashSet;
+
+use crate::graphs::{keys::NodeIdx, Graph};
+
+/// Implementation of the `BFS` algorythm
+///
+/// when `d` is the distance between a node and the startnode,
+/// it will evaluate every node with `d=1`, then continue with `d=2` and so on.
+pub struct BreadthFirstSearch {
+    queue: VecDeque<NodeIdx>,
+    visited: HashSet<NodeIdx>,
+}
+
+impl BreadthFirstSearch {
+    /// Creates a new `DepthFirstSearch` with a start node and the count of nodes
+    pub fn new(start: NodeIdx, count: usize) -> Self {
```

Comment:
Can we make the `count` argument optional? It looks like it's just used as a size hint optimization.

If so, we should be sure to clearly document that.

Reply:
sure, i'll make a dedicated `with_capacity` function and clearer docs